src/components/Trade/TradeInput.tsx

```diff
@@ -47,28 +54,135 @@ export const TradeInput = ({ history }: RouterProps) => {
     number: { localeParts, toFiat },
   } = useLocaleFormatter()
   const [isSendMaxLoading, setIsSendMaxLoading] = useState<boolean>(false)
-  const [quote, buyTradeAsset, sellTradeAsset, feeAssetFiatRate, quoteError] = useWatch({
-    name: ['quote', 'buyAsset', 'sellAsset', 'feeAssetFiatRate', 'quoteError'],
-  }) as [TS['quote'], TS['buyAsset'], TS['sellAsset'], TS['feeAssetFiatRate'], TS['quoteError']]
-  const {
-    updateQuote,
-    checkApprovalNeeded,
-    getSendMaxAmount,
-    updateTrade,
-    feeAsset,
-    refreshQuote,
-  } = useSwapper()
+  const [
```

Comment:
note for the future - we only have to cast here as `react-hook-form` is unable to unpack the types of deeply nested objects.

if we call `useWatch` with a single value on multiple lines, it can infer the types correctly.

i'd previously attempted to fix this in a few places while working in areas like this, but it meant rejigging basically all of our tests that rely on the order and mocking of `useWatch` calls (which is kind of a problem in and of itself).